src/components/SpecialResolution/ResolutionEditor.vue

```diff
@@ -0,0 +1,318 @@
+            need to be amended.
+          </p>
+        </v-col>
+      </v-row>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'pinia-class'
+import {
+  TiptapVuetify,
+  Heading,
+  Bold,
+  Italic,
+  Strike,
+  Underline,
+  BulletList,
+  OrderedList,
+  ListItem,
+  Blockquote,
+  HardBreak,
+  HorizontalRule,
+  History,
+  Table,
+  TableCell,
+  TableHeader,
+  TableRow
+} from 'tiptap-vuetify'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker/'
+import { useStore } from '@/store/store'
+import { ActionBindingIF } from '@/interfaces'
+import { VuetifyRuleFunction } from '@/types'
+import { SpecialResolutionIF } from '@bcrs-shared-components/interfaces'
+import DateUtilities from '@/services/date-utilities'
+
+@Component({
+  components: {
+    DatePickerShared,
+    TiptapVuetify
+  }
+})
+export default class ResolutionEditor extends Vue {
+  @Getter(useStore) getBusinessFoundingDateTime!: string
+  @Getter(useStore) getComponentValidate!: boolean
+  @Getter(useStore) getCurrentDate!: string
+  @Getter(useStore) getSpecialResolution!: SpecialResolutionIF
+
+  @Action(useStore) setSpecialResolution!: ActionBindingIF
+  @Action(useStore) setSpecialResolutionValid!: ActionBindingIF
+
+  $refs!: {
+    resolutionDatePickerRef: DatePickerShared,
+  }
+
+  resolutionDateText = ''
+  resolution = ''
+
+  extensions = [
+    History,
+    Blockquote,
+    Underline,
+    Strike,
+    Italic,
+    ListItem,
+    BulletList,
+    OrderedList,
+    [
+      Heading,
+      {
+        options: {
+          levels: [1, 2, 3]
+        }
+      }
+    ],
+    Bold,
+    HorizontalRule,
+    HardBreak,
+    Table,
+    TableCell,
+    TableHeader,
+    TableRow
+  ]
+
+  editorProperties = {
+    transformPastedText (text) {
+      return text.replaceAll(/[\u200B-\u200D\uFEFF\u200E\u200F]|(?:&#x200E;)/g, '') // eslint-disable-line
+    },
+    transformPastedHTML (html) {
+      return html.replaceAll(/[\u200B-\u200D\uFEFF\u200E\u200F]|(?:&#x200E;)/g, '') // eslint-disable-line
+    }
+  }
+
+  /** Validations rules for resolution date field. */
+  get dateRules (): Array<VuetifyRuleFunction> {
+    return [
+      (v: string) => !!v || 'Resolution date is required',
+      (v: string) =>
+        this.isValidDateRange(this.resolutionDateMin,
+          this.resolutionDateMax,
+          v) ||
+        `Date should be between ${DateUtilities.yyyyMmDdToPacificDate(this.resolutionDateMin, true)} and
+         ${DateUtilities.yyyyMmDdToPacificDate(this.resolutionDateMax, true)}`
+    ]
+  }
+
+  /**
+   * True if date is >= the minimum (ie, today) and <= the maximum (ie, the 10th day).
+   * This is used for Vue form validation (in Date Rules above).
+   */
+  isValidDateRange (minDateStr: string, maxDateStr: string, dateStrToValidate: string): boolean {
```

Comment:
Just a thought here, would you think it's a good idea if we move this function to DateUtilities?

Reply:
I'd rather use a date library - Luxon - https://github.com/moment/luxon/blob/master/docs/why.md 
https://moment.github.io/luxon/demo/global.html

or date-fns and not reinvent the wheel

Reply:
Oh OK. Makes sense. Thanks and resolving!

Reply:
I've read numerous blogs/articles saying that we don't need date libraries (esp moment) with modern JS.

I would prefer to put reusable date functions in the date mixin or utilities class.

I don't think we need anything extensive for date handling. The reasons for our existing date class are:
1. to convert between our 2 or 3 different format (which I should explain some time)
2. to try to have "correct" time in the browser (which runs on a PC that may be in a different time zone or incorrect or spoofed)

That said, Jonathan partially convinced me that we mostly don't care about number 2 because the Legal API will validate things anyways.

We could have a good discussion on this, but maybe later? Travis, up to you whether to move this function. There's pros and cons to both approaches and I'm OK with either.